paper-api/src/main/java/org/bukkit/event/block/LeavesDistanceUpdateEvent.java

```diff
@@ -0,0 +1,76 @@
+package org.bukkit.event.block;
+
+import com.google.common.base.Preconditions;
+import org.bukkit.block.Block;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Called when block distance gets changed.
+ * <p>
+ * If this event is cancelled, the block neighbors will not be updated.
+ */
+public class LeavesDistanceUpdateEvent extends BlockEvent implements Cancellable {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+
+    private final int minDistance;
+    private final int maxDistance;
+    private final int oldDistance;
+    private int newDistance;
+    private boolean cancelled;
+
+    @ApiStatus.Internal
+    public LeavesDistanceUpdateEvent(@NotNull final Block block, final int minDistance, final int maxDistance, final int oldDistance, final int newDistance) {
+        super(block);
+        this.minDistance = minDistance;
+        this.maxDistance = maxDistance;
+        this.oldDistance = oldDistance;
+        this.newDistance = newDistance;
+    }
+
+    public int getMinDistance() {
+        return this.minDistance;
+    }
+
+    public int getMaxDistance() {
+        return this.maxDistance;
+    }
```

Comment:
It feels weird to have these as methods. Aren't these constant values? just the min/max of that specific block state property?

I don't think they need to be exposed to the API like this (especially in lieu of https://github.com/PaperMC/Paper/pull/11963 which lets people access them directly.

I would just have the setter check against a hardcoded min/max in a precondition.